### ISSUE TITLE
HANA - Sepparate users from schemas

### ISSUE THREAD

**Issue by maritogaleano** [dbeaver/dbeaver#11257]
Shows created database users as schema

![image](https://user-images.githubusercontent.com/57814047/107215779-86d06480-69ea-11eb-8b2e-1c1ee064bce3.png)

![image](https://user-images.githubusercontent.com/57814047/107215823-9bacf800-69ea-11eb-98e2-f8fa03914f0c.png)

**Comment by kai-morich** [comment]
not a bug. For each user a corresponding schema is created by HANA

**Comment by maritogaleano** [comment]
![image](https://user-images.githubusercontent.com/57814047/108104042-8b8bbd00-7069-11eb-921b-70e7d16144f4.png)
Analyzing in the native tool that Hana offers, which is SAP Hana Studio, for each user accessing a certain schema, it does not do that.

**Comment by kai-morich** [comment]
looks like HANA Studio filters out schemas where SYS.SCHEMAS.HAS_PRIVILEGES = FALSE, as one may see objects in that schema but no content. 

Could be implemented here as a convenience feature.

**Comment by maritogaleano** [comment]
@kai-morich It would be ideal to include it as an update, because then it becomes somewhat confusing when you have multiple users

**Comment by LonwoLonwo** [comment]
Query example for such non-schemas users will be very helpful.

**Comment by wernerdaehn** [comment]
I have tested all combinations I can think of:

* A user with a schema
* A new schema owned by somebody else (CREATE SCHEMA schema_without_user owned BY dbeaver_normal;)

Dbeaver works as expected.

What Hana Studio does is, similar - but not exactly - to what @kai-morich said, is hide all schemas that have no objects the connected user can see.
In other words, the list of schemas is not inferred from the schemas view (SELECT * FROM schemas;) but from the objects list (SELECT DISTINCT schema_name FROM objects;). User that have no access to any object in a schema or a schema that has no objects are not shown.

Question is, if that is a good idea or not? I feel yes, Dbeaver should hide schemas without objects. Hence I would change the Issue title yet again. ;-)

**Comment by LonwoLonwo** [comment]
I understand correctly, what do you want to see all the schemes except empty?

**Comment by wernerdaehn** [comment]
Correct, that makes the most sense. To show all schemas except empty (based on the permissions of the user). 
The select 
```
SELECT DISTINCT schema_name FROM objects
```
gives you that.

**Comment by kai-morich** [comment]
I would include all schemas, where the user has sufficient privileges.

So instead of 

    SELECT DISTINCT schema_name FROM objects

I recommend to use

    SELECT schema_name FROM sys.schemas WHERE HAS_PRIVILEGES ='TRUE'

But that very likely this is only for display. I guess internally a full list of schemas is required, else something like object dependencies might fail
